File: Python/strategy/quack_behavior.py
from abc import ABC
from abc import abstractmethod


class QuackBehavior(ABC):
    @abstractmethod
    def quack(self):
        pass


class Quack(QuackBehavior):
    def quack(self):
        print("꽥꽥!")


class Squeak(QuackBehavior):
    def quack(self):
        print("삑삑~")


class MuteQuack(QuackBehavior):
    def quack(self):
        print("silence")

File: Kotlin/src/main/kotlin/chapter1/QuackBehavior.kt
package chapter1

interface QuackBehavior {
    fun quack()
}

class Quack: QuackBehavior {
    override fun quack() {
        println("꽥")
    }
}

class MuteQuack: QuackBehavior {
    override fun quack() {
        println("조용")
    }
}

class Squeak: QuackBehavior {
    override fun quack() {
        println("삑")
    }
}